apps/web/src/services/cost-explorer/cost-analysis/modules/CostAnalysisPeriodSelectDropdown.vue

```diff
@@ -114,6 +123,38 @@ const state = reactive({
 });
 
 /* Util */
+const getPeriodItemNameByRelativePeriod = (relativePeriod?: RelativePeriod) => state.allPeriodItems.find((item) => isEqual(item.relativePeriod, relativePeriod))?.name;
+const setSelectedItemByOptions = ({ relativePeriod, period, granularity }:ParamsForSelectedPeriod) => {
+    console.log('setSelectedItemByOptions', { relativePeriod, period, granularity });
+    if (!relativePeriod && !period && granularity) {
+        const [defaultPeriod, defaultRelativePeriod] = initiatePeriodByGranularity(granularity);
+        costAnalysisPageStore.$patch((_state) => {
+            _state.period = defaultPeriod;
+            _state.relativePeriod = defaultRelativePeriod;
+        });
+        if (granularity === GRANULARITY.DAILY) {
+            state.selectedPeriod = today.subtract(0, 'month').format('YYYY-MM');
+        } else {
+            console.log('어때', getPeriodItemNameByRelativePeriod(defaultRelativePeriod), defaultRelativePeriod);
+            state.selectedPeriod = getPeriodItemNameByRelativePeriod(defaultRelativePeriod);
+        }
+        return;
+    }
+    if (granularity) {
+        costAnalysisPageStore.$patch((_state) => {
+            _state.granularity = granularity;
+        });
+    }
+    if (relativePeriod) {
+        console.log('setSelectedItemByOptions', getPeriodItemNameByRelativePeriod(relativePeriod), relativePeriod);
+        state.selectedPeriod = getPeriodItemNameByRelativePeriod(relativePeriod);
+    } else if (granularity === GRANULARITY.DAILY) {
+        const selectedPeriodItem:PeriodItem|undefined = state.dailyPeriodItems.find((item) => isEqual(item.period, period));
+        state.selectedPeriod = selectedPeriodItem?.name;
+    } else {
+        state.selectedPeriod = 'custom';
+    }
+};
```

Comment:
Suggestion: How about adding annotation to separate each case? for example 'Case by selecting granularity with dropdown'... And maybe you know, there are `console.log`!